Balance-Druid-improve-Starfall-usage-and-add-CC-safety

gacuna89 opened the pull request:
- Move Starfall from default actions to AOE strategy only
- Require 2+ enemies for Starfall usage (prevents single-target casting)
- Add CC safety: avoid casting Starfall near current CC targets
- Prioritize Starfall over Hurricane in medium AOE situations
- Remove Starfall from pull/opener rotation to prevent early single-target usage

This prevents Balance druids from wasting Starfall on single targets
and breaking crowd control effects in group content.

Dohtt commented:
I do feel like starfall has a crucial place in a druids normal kit for bossfights.
<img width="871" height="392" alt="image" src="https://github.com/user-attachments/assets/ca55e5d5-8135-4164-b3af-f5b4ba253ca6" />


It's also listed as a high prio spell, besides during eclipse procs and should be on permanent cooldown. 

I think this might reduce the balance druids dps capabilities.

gacuna89 commented:
> I do feel like starfall has a crucial place in a druids normal kit for bossfights. <img alt="image" width="871" height="392" src="https://private-user-images.githubusercontent.com/17968897/497951678-ca55e5d5-8135-4164-b3af-f5b4ba253ca6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk3Nzk2NDYsIm5iZiI6MTc1OTc3OTM0NiwicGF0aCI6Ii8xNzk2ODg5Ny80OTc5NTE2NzgtY2E1NWU1ZDUtODEzNS00MTY0LWIzYWYtZjViNGJhMjUzY2E2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDA2VDE5MzU0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc3NDcwZmQ1MzU2ZTE3MDBkYmZiMjAyMGQzMzVjZDliZGVmNzU4ZDM1NzIzYTAyMmQzZGEwNDI0OTg2M2RhMjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KDo4GioOqo14ZQWHUJ5pAnmOhvsUdUkogzjOd5MsPxc">
> 
> It's also listed as a high prio spell, besides during eclipse procs and should be on permanent cooldown.
> 
> I think this might reduce the balance druids dps capabilities.

You're right, I was being too strict with Starfall.
After checking the rotation guide again, it should just be used on cooldown as a high-priority spell outside Eclipse.

I tweaked the logic a bit to make it more balanced:

Kept:
- Starfall back in the main actions with high priority (keeps DPS flow).
- Still avoids casting near CC’d targets to not break anything.

Changed:
- Removed the AOE-only restriction that was hurting damage.
- Only blocks single-target use for the first few seconds of combat (so it’s not wasted right at the start).
- After that, it just works normally on cooldown.

That should give a solid rotation without breaking CC or wasting Starfall on single-target fights.

hermensbas commented:
Does this still require testing?
